rerun_cpp/src/rerun/status.hpp

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#ifdef __cpp_exceptions
+#include <stdexcept>
+#endif
+
+struct rr_status;
+
+namespace rerun {
+    /// Status codes returned by the SDK as part of `Status`.
+    ///
+    /// Category codes are used to group errors together, but are never returned directly.
+    enum class StatusCode : uint32_t {
+        Ok = 0,
+
+        // Invalid argument errors.
+        _CategoryArgument = 0x000000010,
+        UnexpectedNullArgument,
+        InvalidStringArgument,
+        InvalidRecordingStreamHandle,
+        InvalidSocketAddress,
+        InvalidEntityPath,
+
+        // Recording stream errors
+        _CategoryRecordingStream = 0x000000100,
+        RecordingStreamCreationFailure,
+        RecordingStreamSaveFailure,
+
+        // Arrow data processing errors.
+        _CategoryArrow = 0x000001000,
+        ArrowIpcMessageParsingFailure,
+        ArrowDataCellError,
+
+        Unknown = 0xFFFFFFFF,
+    };
+
+    /// Callback function type for log handlers.
+    using StatusLogHandler = void (*)(const class Status& status, void* userdata);
+
+    /// Status outcome object (success or error) returned for fallible operations.
+    ///
+    /// Converts to `true` for success, `false` for failure.
+    class [[nodiscard]] Status {
+      public:
+        /// Result code for the given operation.
+        StatusCode code = StatusCode::Ok;
+
+        /// Human readable description of the error.
+        std::string description;
+
+      public:
+        Status() = default;
+
+        Status(StatusCode _code, std::string _description)
+            : code(_code), description(std::move(_description)) {}
+
+        /// Construct from a C status object.
+        Status(const rr_status& status);
+
+        operator bool() const {
```

Comment:
`operator bool` is not good - it also means `Status` is implicitly converted to an int, because C++

https://www.kdab.com/explicit-operator-bool/

```suggestion
        operator void*() const {
```